test/spec/grammar/dsl_api_test.js

```diff
@@ -0,0 +1,140 @@
+const expect = require('chai').expect;
+const tokens = require('../../../lib/dsl/lexer').tokens;
+const getSyntacticAutoCompleteSuggestions = require('../../../lib/dsl/api')
+  .getSyntacticAutoCompleteSuggestions;
+const parse = require('../../../lib/dsl/api')
+  .parse;
+
+
+describe('JDL DSL API', () => {
+  describe('when wanting an AST', () => {
+    describe('with a valid input', () => {
+      it('returns an AST', () => {
+        const ast = parse(`
+          entity A {
+            field String,
+          }`);
+        expect(ast.entities).to.have.lengthOf(1);
+        expect(ast.entities[0]).to.deep.eql({
+          name: 'A',
+          tableName: 'A',
+          body: [
+            {
+              name: 'field',
+              type: 'String',
+              validations: [],
+              javadoc: null
+            }
+          ],
+          javadoc: null
+        });
+      });
+    });
+
+    describe('with a lexing error', () => {
+      it('throws an error', () => {
+        const invalidInput = 'entity ± {';
+        expect(() => parse(invalidInput)).to
+          .throw('unexpected character: ->±<- at offset: 7, skipped 1 characters.');
+      });
+    });
+
+    describe('with a Parsing error', () => {
+      describe('with a MismatchedTokenException', () => {
+        it('throws an error', () => {
+        // wrong type of closing parenthesis ']'
+          const invalidInput = 'entity Person { ]';
+          expect(() => parse(invalidInput)).to
+            .throw('MismatchedTokenException: Expecting --> \'}\' <-- but found --> \']\' <--\n' +
+            '\tat line: 1, column: 17');
+        });
+      });
+
+      describe('with a MismatchedTokenException for EOF', () => {
+        it('throws an error', () => {
+          // No Closing parenthesis
+          const invalidInput = 'entity Person {';
+          expect(() => parse(invalidInput)).to
+            .throw('MismatchedTokenException: Expecting --> \'}\' <-- but found --> \'\' <--');
+        });
+      });
+    });
+
+    describe('with a semantic validation error', () => {
+      it('throws an error', () => {
+        // lower case entityName first char
+        const invalidInput = 'entity person { }';
+        expect(() => parse(invalidInput)).to
+          .throw('Entity name must match: /^[A-Z][A-z0-9]*$/\n' +
+            '\tat line: 1, column: 8');
+      });
+    });
+  });
+
+  describe('when wanting an auto-completion', () => {
+    describe('with an empty text', () => {
+      it('provides suggestions', () => {
+        const input = '';
+        const result = getSyntacticAutoCompleteSuggestions(input);
+        expect(result).to.have.lengthOf(17);
+        expect(result).to.have.members([
```

Comment:
For L.79 & L80: consider replacing them by `deep.equal`. I don't know if it's applicable here, just suggesting.

Reply:
That is a good suggestion, but that is not part of code of this commit. (renamed file)

Reply:
I tested it, it won't work because deep.equal tests the order of elements as well
and when building the expected list of token via the grammar
my brain and the algorithm used in Chevrotain output a different order of elements 😄